pkg/build/strategies/dockerfile/dockerfile.go

```diff
@@ -153,7 +148,8 @@ func (builder *Dockerfile) CreateDockerfile(config *api.Config) error {
 			buffer.WriteString("# Copying in override save-artifacts script\n")
 			artifactsScript = sanitize(filepath.ToSlash(filepath.Join(scriptsDestDir, "save-artifacts")))
 			uploadScript := sanitize(filepath.ToSlash(filepath.Join(builder.uploadScriptsDir, "save-artifacts")))
-			buffer.WriteString(fmt.Sprintf("COPY --chown=%s:0 %s %s\n", sanitize(imageUser), uploadScript, artifactsScript))
+			buffer.WriteString(fmt.Sprintf("COPY %s %s\n", uploadScript, artifactsScript))
+			buffer.WriteString(fmt.Sprintf("RUN chown %s:0 %s\n", sanitize(imageUser), artifactsScript))
```

Comment:
Run `chown` on the `save-artifacts` script that was copied into the incremental build image.